subxt/src/events/mod.rs

```diff
@@ -14,7 +14,63 @@
 // You should have received a copy of the GNU General Public License
 // along with subxt.  If not, see <http://www.gnu.org/licenses/>.
 
-//! For working with events.
+//! Parse the events generated by a block.
+//!
+//! This module exposes the ability to work with events generated by a given block.
+//! There are two types of events that the subxt is internally aware of:
+//! - static events: Events that are already present in the subxt's client metadata
+//! - dynamic events: Events that are present only in the target node's metadata
+//!
+//! For the later case can happen when the subxt connects first to a target node,
+//! then the target node performs a runtime upgrade. To mitigate this behavior and keep the
+//! subxt's metadata in sync with the target node, please refer to [crate::updates::UpdateClient].
+//!
+//! This module is wrapped by the generated API in `RuntimeAPI::EventsApi`.
+//!
```

Comment:
```suggestion
```

I'd be inclined to just remove this; the comemnt isn't quite true. If a node updates and event shapes change, then static decoding will simply fail, rather than hand back dynamic event details :)